src/indexable.cr

```diff
@@ -412,24 +412,80 @@ module Indexable(T)
     indexes.map { |index| self[index] }
   end
 
+  # Calls the given block once for each index in `self`, passing `self`
```

Comment:
`` `self` and `other` index element`` is ambiguous and difficult to understand. Maybe `` `{self[i], other[i]}` for each index``?